server/command_test.go

```diff
@@ -238,6 +239,132 @@ func TestExecuteShowCommand(t *testing.T) {
+					Message:   "No links present.",
+				}).Return(testutils.GetPost(testutils.GetChannelID(), testutils.GetUserID())).Once()
+			},
+			setupStore: func(s *mockStore.Store) {
+				s.On("GetLinks").Return(nil, nil).Times(1)
+			},
+			setupClient: func(c *mockClient.Client) {},
+		},
+		{
+			description: "Error in fetching info from MS Teams",
+			args: &model.CommandArgs{
+				UserId:    testutils.GetUserID(),
+				ChannelId: testutils.GetChannelID(),
+			},
+			setupAPI: func(api *plugintest.API) {
+				api.On("GetTeam", testutils.GetTeamID()).Return(nil, testutils.GetInternalServerAppError("error in getting team info")).Times(4)
+				api.On("GetChannel", testutils.GetChannelID()).Return(nil, testutils.GetInternalServerAppError("error in getting channel info")).Times(4)
+
+				api.On("LogDebug", "Unable to get the MS Teams teams information", "Error", "error in getting teams info").Once()
+				api.On("LogDebug", "Unable to get the MS Teams channel information for a team", "TeamID", testutils.GetTeamsTeamID(), "Error", "error in getting channels info").Once()
+				api.On("LogDebug", "Unable to get the Mattermost team information", "TeamID", testutils.GetTeamID(), "Error", "error in getting team info").Times(4)
+				api.On("LogDebug", "Unable to get the Mattermost channel information", "ChannelID", testutils.GetChannelID(), "Error", "error in getting channel info").Times(4)
+
+				api.On("SendEphemeralPost", testutils.GetUserID(), &model.Post{
+					UserId:    "bot-user-id",
+					ChannelId: testutils.GetChannelID(),
+					Message:   "Please wait while your request is being processed.",
+				}).Return(testutils.GetPost(testutils.GetChannelID(), testutils.GetUserID())).Once()
+
+				api.On("SendEphemeralPost", testutils.GetUserID(), &model.Post{
+					UserId:    "bot-user-id",
+					ChannelId: testutils.GetChannelID(),
+					Message:   "| Mattermost Team | Mattermost Channel | MS Teams Team | MS Teams Channel | \n| :------|:--------|:-------|:-----------|\nThere were some errors while fetching information about some links. Please check the server logs.",
+				}).Return(testutils.GetPost(testutils.GetChannelID(), testutils.GetUserID())).Once()
+			},
+			setupStore: func(s *mockStore.Store) {
+				s.On("GetLinks").Return(testutils.GetChannelLinks(4), nil).Times(1)
+			},
+			setupClient: func(c *mockClient.Client) {
+				c.On("GetTeams", mock.AnythingOfType("string")).Return(nil, errors.New("error in getting teams info")).Times(4)
+				c.On("GetChannelsInATeam", testutils.GetTeamsTeamID(), mock.AnythingOfType("string")).Return(nil, errors.New("error in getting channels info")).Times(4)
+			},
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			testCase.setupAPI(mockAPI)
+			p.SetAPI(mockAPI)
+			defer mockAPI.AssertExpectations(t)
+
+			testCase.setupStore(p.store.(*mockStore.Store))
+			testCase.setupClient(p.msteamsAppClient.(*mockClient.Client))
+			_, _ = p.executeShowLinksCommand(testCase.args)
```

Comment:
check this error as well

Reply:
There's no use because we are already checking the error returned with the function "SendEphemeralPost" and this function always returns the error as "nil".